shoes-swt/lib/shoes/swt/arrow_painter.rb

```diff
@@ -0,0 +1,36 @@
+class Shoes
+  module Swt
+    class ArrowPainter < Common::Painter
+      def fill(gc)
+        gc.fill_path(path)
+      end
+
+      def draw(gc)
+        gc.draw_path(path)
+      end
+
+      def path
+        @path ||= begin
+          path = ::Swt::Path.new(::Swt.display)
+
+          path.move_to(@obj.left - @obj.width / 2, @obj.top)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top - @obj.width * 0.20 - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.50, @obj.top)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top + @obj.width * 0.20 + @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top + @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top + @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top)
```

Comment:
So was actually underway making a stab at giving the positions names at least rather than the constants. Ran with body/head naming and gave a diagram to help illustrate where the various points are located. What do you think does that help?

On plus side, passes Rubocop without modification now :+1: